core/src/time.rs

```diff
@@ -124,6 +125,9 @@ pub trait DurationExt: Sized {
     /// Return a duration representing this duration rounded up to the next largest multiple of
     /// `time_precision`, or the same duration if it already a multiple.
     fn round_up(&self, time_precision: &Duration) -> Result<Self, Error>;
+
+    /// Confirm that this duration is a multiple of the task time precision.
+    fn validate_precision(self, time_precision: &Duration) -> Result<Self, janus_messages::Error>;
```

Comment:
I think this type is already imported as `Error`

```suggestion
    fn validate_precision(self, time_precision: &Duration) -> Result<Self, Error>;
```